third_party/move/move-compiler-v2/tests/checking-lang-v2.2/lambda/fun_param_variance.move

```diff
@@ -0,0 +1,12 @@
+// Fixes ##16403
+module 0x66::xx {
+    fun apply(f: |&mut u64|) {
+        let x = 0;
+        f(&mut x);
+    }
+
+    fun test() {
+        let func = |x: &u64| { *x + 1; };
+        apply(func);
+    }
+}
```

Comment:
Would be good to add the other test case (one dealing with return type variance) mentioned in #16403 as well, here:

```
module 0x66::yy {
    fun apply(f: |&mut u64|&u64) {
        f(&mut 0);
    }

    fun test() {
        let func: |&mut u64|&mut u64 = |x| { x };
        apply(func);
    }
}
```

Reply:
Done